### PR TITLE
Web3 integration Pt.1

### PR DESCRIPTION
**Pull request by gpablo6**
Includes:
- Restructure of project directory
- Change of Web3 React-Interface
- Implementation of Contract Functionalities
- Implementation of components for Donate contract

Pending:
- Minor fixed to Escrow functions
- Implementation of components for Escrow contract

### CODE REVIEW

File: scripts/donations.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as ethers from "ethers";
+
+export function getDonateContract() {
+    const contractAddress = "0xE1bF07E88D873E943755595E5401DCB222eF4725"
```
Comment (rodrigopavezi) [inline scripts/donations.ts:4]: I think this contract address should go the the .env so you can switch between test and main nets. 

More info on how to do it here: https://nextjs.org/docs/basic-features/environment-variables

**Comment by lalomartins** [comment]
Can you please undo the repo restructuring. It's a good idea, but it should never be done inside such a big branch, as it makes the PR more or less impossible to review and very difficult to merge. We can restructure the repo separately after this is merged.

**Review by lalomartins — CHANGES_REQUESTED** [review]
Can you please undo the repo restructuring. It's a good idea, but it should never be done inside such a big branch, as it makes the PR more or less impossible to review and very difficult to merge. We can restructure the repo separately after this is merged.

**Comment by lalomartins** [comment]
The backend branch is now merged to main, so when this branch is building, we can merge it too. Let me know 😸

**Pull request closed without merging**